### PR TITLE
docs(testing): clarify apps & libs can be component tested

### LINKED ISSUE #12960: Component testing guidelines/best practices
### Documentation issue

- [ ] Reporting a typo
- [ ] Reporting a documentation bug
- [x] Documentation improvement
- [ ] Documentation feedback

### Is there a specific documentation page you are reporting?

https://nx.dev/cypress/cypress-component-testing

### Additional context or description

I am trying to setup component testing in our NX workspace. Many of our apps are very thin and imports components from libraries. The component libraries include all the associated business/UI logic. 

There is not much guidance on where you should put your component tests. I was thinking of putting in my library. But our libraries are not buildable and cypress needs a `devServerTarget`. We use a custom webpack executor to build our apps and copied that into our lib to make it work. Im not sure if this is the right approach.

It would also be great if the docs were improved for custom webpack users. [This is an issue](https://github.com/nrwl/nx/issues/12844) I filed to figure how to get it working.

### PR DESCRIPTION
**Pull request by barbados-clemens**
## Current Behavior

unsure what I can CT
## Expected Behavior

explain both apps and libs should work

## Related Issue(s)

Fixes #12960

### CODE REVIEW

**Review by bcabanes — APPROVED** [review]

File: docs/generated/packages/cypress/documents/cypress-component-testing.md [hunk 1]
```diff
@@ -22,6 +22,8 @@ You can optionally pass in `--generate-tests` to create component tests for all
 
 Component testing leverages a build target within your workspace as the base for running the tests. The build target is usually an app within the workspace. By default, the generator attempts to find the build target for you based on the project usage, but you can manually specify the build target to use via the `--build-target` option.
 
+The configuration generators support both application and library types, so while a `--build-target` flag might be needed, it does not need to the build target of the same project you are trying to add component testing too.
```
Comment (isaacplmann) [inline docs/generated/packages/cypress/documents/cypress-component-testing.md:25]:
```suggestion
The configuration generators support both application and library types, so while a `--build-target` flag might be needed, it does not need to be the build target of the same project you are trying to add component testing too.
```

File: docs/shared/packages/cypress/cypress-component-testing.md [hunk 2]
```diff
@@ -22,6 +22,8 @@ You can optionally pass in `--generate-tests` to create component tests for all
 
 Component testing leverages a build target within your workspace as the base for running the tests. The build target is usually an app within the workspace. By default, the generator attempts to find the build target for you based on the project usage, but you can manually specify the build target to use via the `--build-target` option.
 
+The configuration generators support both application and library types, so while a `--build-target` flag might be needed, it does not need to the build target of the same project you are trying to add component testing too.
```
Comment (isaacplmann) [inline docs/shared/packages/cypress/cypress-component-testing.md:25]:
```suggestion
The configuration generators support both application and library types, so while a `--build-target` flag might be needed, it does not need to be the build target of the same project you are trying to add component testing too.
```

**Review by isaacplmann — CHANGES_REQUESTED** [review]
typo

**Review by isaacplmann — APPROVED** [review]

File: docs/generated/packages/cypress/documents/cypress-component-testing.md [hunk 3]
```diff
@@ -22,6 +22,8 @@ You can optionally pass in `--generate-tests` to create component tests for all
 
 Component testing leverages a build target within your workspace as the base for running the tests. The build target is usually an app within the workspace. By default, the generator attempts to find the build target for you based on the project usage, but you can manually specify the build target to use via the `--build-target` option.
 
+The configuration generators support both application and library types, so while a `--build-target` flag might be needed, it does not need to be the build target of the same project you are trying to add component testing too.
```
Comment (FrozenPandaz) [inline docs/generated/packages/cypress/documents/cypress-component-testing.md:25]:
```suggestion
Component testing supports both applications and libraries. By default, the generator attempts to find the build target for you based on the project's dependent apps. But you can manually specify the build target to use via the `--build-target` option. Note, in most cases, the build target will be from a different project than the one being configured. The only case where the build targets are from the same project is when the component tests are being added to an application.
```

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**